x/staking/types/params.go

```diff
@@ -203,3 +210,48 @@ func validateMinCommissionRate(i interface{}) error {
 
 	return nil
 }
+
+func validateMaxCommissionRate(i interface{}) error {
+	v, ok := i.(math.LegacyDec)
+	if !ok {
+		return fmt.Errorf("invalid maximum parameter type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("minimum commission rate cannot be nil: %s", v)
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("minimum commission rate cannot be negative: %s", v)
+	}
+	if v.GT(math.LegacyOneDec()) {
+		return fmt.Errorf("minimum commission rate cannot be greater than 100%%: %s", v)
+	}
+
+	return nil
+}
+
+func validateCommissionRate(minimum, maximum interface{}) error {
+	if err := validateMinCommissionRate(minimum); err != nil {
+		return err
+	}
+
+	if err := validateMaxCommissionRate(maximum); err != nil {
+		return err
+	}
```

Comment:
Let's have a single `validateCommissionRate()` for both min and max, and then wrap any error to indicate if the related commision was min or max.

```go
if err := validateCommissionRate(minimum); err != nil {
		return fmt.Error("min commission: %w", err)
}
if err := validateCommissionRate(maximum); err != nil {
		return fmt.Error("max commission: %w", err)
}
```